core/tee/tee_svc_cryp.c

```diff
@@ -3420,7 +3422,12 @@ TEE_Result tee_svc_asymm_verify(uint32_t state,
 	if (res != TEE_SUCCESS)
 		goto out;
 
-	if (data_len != hash_size) {
+    	/*
+    	 * Depending on the DSA algorithm, the size of the hash to consider
+	 * could be less or equal than he whole size of the
+	 * hash algorithm involved (related to the subprime).
```

Comment:
Is the subprime something like the modulus in RSA, that is, defining the size of the result of the operation?

Please update the comment with that clarification, also a short note on why it does not affect the security of the other algorithms would nice.